docs/source/core_specification/schema_types.rst

```diff
@@ -1,3 +1,282 @@
 ============
-.. TODO: Explain how to print a schema in Schema Types.
+
+A GraphQL schema might look like the one below. Do not be intimidated by the number of components
+since we will immediately proceed to dissect the schema.
+
+.. TODO: Use a better "documentation" schema. I used a subset of the schema that we used in tests
+   because it was the one referenced by all the queries in the Directives section and I can
+   easily modify  the directives section so that it only includes types in this subset. However,
+   it is  a bit more difficult to completely change what schema we are using for documentation.
+   Though this schema is less than ideal for documentation. It is to large and some of types,
+   like Entity, are not intuitive.
+
+.. code::
+
+    schema {
+        query: RootSchemaQuery
+    }
+
+    type RootSchemaQuery {
+        Animal: [Animal]
+        Entity: [Entity]
+        Food: [Food]
+        Species: [Species]
+        Toy: [Toy]
+    }
+
+    directive @filter(op_name: String!, value: [String!]) on FIELD | INLINE_FRAGMENT
+
+    directive @tag(tag_name: String!) on FIELD
+
+    directive @output(out_name: String!) on FIELD
+
+    directive @output_source on FIELD
+
+    directive @optional on FIELD
+
+    directive @recurse(depth: Int!) on FIELD
+
+    directive @fold on FIELD
+
+    scalar Date
+
+    scalar DateTime
+
+    scalar Decimal
+
+    type Animal implements Entity {
+        name: String
+        alias: [String]
+        birthday: Date
+        net_worth: Decimal
+        in_Animal_ParentOf: [Animal]
+        out_Animal_ParentOf: [Animal]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        out_Animal_OfSpecies: [Species]
+        out_Animal_PlaysWith: [Toy]
+    }
+
+    type Food implements Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        in_Species_Eats: [Species]
+    }
+
+    type Species implements Entity {
+        name: String
+        alias: [String]
+        description: String
+        in_Animal_OfSpecies: [Animal]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        in_Species_Eats: [Species]
+        out_Species_Eats: [Union__Food__Species]
+    }
+
+    type Toy {
+        name: String
+        in_Animal_PlaysWith: [Animal]
+    }
+
+    interface Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+    }
+
+    union Union__Food__Species = Food | Species
+
+
+.. note::
+
+    A GraphQL schema can be serialized with the :code:`print_schema` function from the
+    :code:`graphql.utils.schema_printer` module.
+
+
+Objects and Fields
+--------------------------
+
+The core components of a GraphQL schema are GraphQL object types. They conceptually represent the
+concrete vertices in the underlying database. For relational databases, we think of the tables as
```

Comment:
the rows in the tables are vertices, not the tables themselves

Reply:
Now using the term "vertex types".